src/spec.md

```diff
@@ -0,0 +1,237 @@
+
+**Description**: Asynchronously writes data to the EthStorage network.
+
+**Parameters**
+- `key` (string): The key for the data to be written.
+- `data` (Buffer): The data to be written, its size cannot exceed the maximum value of the content that can be transferred by a blob.
+
+**Example**
+```javascript
+const dataToWrite = Buffer.from("some data");
+await ethStorage.write("dataKey", dataToWrite);
+```
+
+#### read
+
+**Description**: Read data asynchronously from the EthStorage network through key.
+
+**Parameters**
+- `key` (string): The key for the data to be read.
+- `ethStorageRpc` (string): RPC of EthStorage network, because the data is obtained from the EthStorage network.
+
+**Returns**
+- `data` (Buffer): The content.
+
+**Example**
+```javascript
+const ethStorageRpc = "https://xxx.rpc";
+const data = await ethStorage.read("example.txt", ethStorageRpc);
+```
+
+
+<p id="FlatDirectory"></p>
+
+## 5. FlatDirectory Class
+
+### Constructor
+
+**Description**: Create an instance of `FlatDirectory`.
+
+**Parameters**
+- `config` (object): Configuration object containing necessary settings.
+    - `rpc` (string): RPC for any evm network.
+    - `privateKey` (string): Wallet private key.
+    - `address` (string, optional): FlatDirectory contract address. If it does not exist, the `deploy` method can be called to create one.
+
+**Example**
+```javascript
+const config = {
+   rpc: "your_rpc",
+   privateKey: "your_private_key",
+   address: "flat_directory_address"
+};
+
+const flatDirectory = new FlatDirectory(config);
+```
+
+### Methods
+
+#### upload
+
+**Description**: Upload data of arbitrary size.
+
+**Parameters**
+- `key` (string): The key of the data.
+- `data` (Buffer): The data to be uploaded.
+- `callbacks` (object): An object containing callback functions:
+  - `onProgress` (function): Callback function that receives `(progress)`, where `progress` is an object containing `count` and `totalCount`.
+  - `onFail` (function): Callback function that receives `(error)`.
+  - `onSuccess` (function): Callback function that receives `()`.
+
+**Example**
+```javascript
+const key = "example1.txt";
+const data = Buffer.from("large data to upload");
+
+await ethStorage.upload(key, data, {
+    onProgress: function(progress) {
+        console.log(`Uploaded ${progress.count} of ${progress.totalCount} chunks`);
+    },
+    onFail: function(error) {
+        console.error("Error uploading data:", error);
+    },
+    onSuccess: function() {
+        console.log("Data uploaded success.");
+    }
+});
+```
+
+#### download
+
+**Description**: Download data by key.
+
+**Parameters**
+- `key` (string): The key for the data to be read.
+- `ethStorageRpc` (string): RPC of EthStorage network, because the data is obtained from the EthStorage network.
+
+**Returns**
+- `data` (Buffer): The content.
+
+**Example**
+```javascript
+const ethStorageRpc = "https://xxx.rpc";
+
+const data = await ethStorage.download("example.txt", ethStorageRpc);
```

Comment:
Why doesn’t the download function have a callback function to track the progress?